backend/main.go

```diff
@@ -185,3 +196,47 @@ func (w *Worker) processStellarWithdrawalRequest(sr store.SignatureRequest) erro
 
 	return nil
 }
+
+func (w *Worker) processEthereumRefundRequest(sr store.SignatureRequest) error {
+	if sr.DepositChain != store.Ethereum {
+		return fmt.Errorf("deposits from %v are not supported", sr.DepositChain)
+	}
+	deposit, err := w.Store.GetEthereumDeposit(context.TODO(), sr.DepositID)
+	if err != nil {
+		return errors.Wrap(err, "error getting ethereum deposit")
+	}
+
+	if err = w.EthereumRefundValidator.CanRefund(context.TODO(), deposit); err != nil {
+		return errors.Wrap(err, "error validating refund conditions")
+	}
+
+	amount, ok := new(big.Int).SetString(deposit.Amount, 10)
+	if !ok {
+		return errors.Errorf("cannot convert value in wei to bit.Rat: %s", deposit.Amount)
+	}
+
+	expiration := int64(math.MaxInt64)
+	sig, err := w.EthereumSigner.SignWithdrawal(
+		common.HexToHash(deposit.ID),
+		expiration,
+		common.HexToAddress(deposit.Sender),
+		common.HexToAddress(deposit.Token),
+		amount,
+	)
+	if err != nil {
+		return errors.Wrap(err, "error signing refund")
+	}
+
+	err = w.Store.UpsertEthereumSignature(context.TODO(), store.EthereumSignature{
+		Address:    w.EthereumSigner.Address().String(),
+		Signature:  hex.EncodeToString(sig),
+		Action:     sr.Action,
+		DepositID:  sr.DepositID,
+		Expiration: expiration,
+	})
```

Comment:
Wanted to double check: the refund in smart contract also requires the signature of the depositor, correct? The `Address` allows them to refund to another address?

Reply:
it does not require the signature of the depositor because the validator signs a message which includes the address which will receive the funds. A few lines above:

```
	sig, err := w.EthereumSigner.SignWithdrawal(
		common.HexToHash(deposit.ID),
		expiration,
		common.HexToAddress(deposit.Sender),
		common.HexToAddress(deposit.Token),
		amount,
	)
```

we specify the recipient as `deposit.Sender`. Therefore this signature can only be used on calls to the smart contract which transfer the funds to `deposit.Sender`.

The `Address` field in store.EthereumSignature is the address of the validator itself. The reason we include that in the response back to the client is because the client will need to present the signatures from the validators to the smart contract in sorted order.